src/nms.py

```diff
@@ -172,32 +178,34 @@ def is_api_available(self) -> bool:
     def login(self, username: str, password: str) -> LoginResponse | None:
         """Login to NMS by sending the username and password and return a Token."""
         login_params = LoginParams(username=username, password=password)
-        response = self._make_request("POST", "/login", data=asdict(login_params))
-        if response:
+        try:
+            response = self._make_request("POST", "/login", data=asdict(login_params))
             return LoginResponse(
                 token=response.get("token"),
             )
-        return None
+        except NMSError:
+            return None
 
     def token_is_valid(self, token: str) -> bool:
         """Return if the token is still valid by attempting to connect to an endpoint."""
-        response = self._make_request("GET", f"/{ACCOUNTS_URL}", token=token)
-        return response is not None
+        try:
+            return self._make_request("GET", f"/{ACCOUNTS_URL}", token=token)
+        except NMSError:
+            return False
 
     def get_status(self) -> StatusResponse | None:
```

Comment:
With the changes below this is not true anymore.

Reply:
BTW, after updating the return type, we can probably remove the `StatusResponse` class.

Reply:
I fixed the issue by creating the `StatusResponse` object. I also added test coverage for those functions that were not covered at all.